### ISSUE TITLE
v0.7.0-rc4 Unable to read previously node config

### ISSUE THREAD

**Issue by Linicks** [input-output-hk/jormungandr#1061]
**While trying to test v0.7.0-rc4 I was unable to start the with my previously working node configuration.**

> ./jormungandr --config ./node-config.yaml --genesis-block-hash ae57995b8fe086ba590c36dc930f2aa9b52b2ffa92c0698fff2347adafe8dc65
> 
> > Error in the overall configuration of the node
> >  |-> Error while parsing the node configuration file: p2p.trusted_peers[0].id: Odd number of digits at line 15 column 9
> >  |-> p2p.trusted_peers[0].id: Odd number of digits at line 15 column 9
> > 

```yaml
---
log:
 format: plain
 level: info
 output: stderr
p2p:
  topics_of_interest:
    blocks: normal
    messages: low
  public_address: "/ip4/ip-address/tcp/3000"
  trusted_peers:
  - address: "/ip4/13.230.137.72/tcp/3000"
    id: ed25519_pk1w6f2sclsauhfd6r9ydgvn0yvpvg4p3x3u2m2n7thknwghrfpdu5sgvrql9
  - address: "/ip4/13.230.48.191/tcp/3000"
    id: ed25519_pk1lzrdh0pcmhwcnqdl5cgcu7n0c76pm7g7p6pdey7wup54vz32gy6qlz5vnq
  - address: "/ip4/18.196.168.220/tcp/3000"
    id: ed25519_pk1uufkgu0t9xm8ry04wnddtnku5gjg8typf5z6ehh65uc6nz4j8n4spq0xrl
  - address: "/ip4/3.124.132.123/tcp/3000"
    id: ed25519_pk14tqkqnz3eydn0c8c8gmmyzxgnf2dztpy5dnrx09mhfzv0dh93s3qszqgpc
  - address: "/ip4/18.184.181.30/tcp/3000"
    id: ed25519_pk178ge2jn6c40vgmrewgmg26nmtda47nk2jncukzj327mp3a9g2qzss2d44f
  - address: "/ip4/184.169.162.15/tcp/3000"
    id: ed25519_pk1nk0ne8ez66w5tp2g8ctcakthjpz89eveyg0egcpylenhet83n0sq2jqz8q
  - address: "/ip4/13.56.87.134/tcp/3000"
    id: ed25519_pk1ce450zrtn04eaevcn9csz0thpjuhxrysdrq6qlr9pq7e0wd842nsxy6r5k
rest:
  listen: ip-address:3101
storage: "./storage"```

**Comment by rdlrt** [comment]
It clearly states in release notes, its not compatible...

**Comment by Linicks** [comment]
@rdlrt I understand that they mentioned the change in the release notes, but haven’t provided any trusted peer id’s, etc, yet.  I was hoping that someone would provide that information so that I could update my configs and spin up my node.

**Comment by mmahut** [comment]
@Linicks please keep in mind trusted peers you mentioned is just some testnet setup by IOHK and not an integral part of this software. The trusted peers you listed are running rc3, thus incompatible. You can run your own network or wait for the new nightly network to be setup in the upcoming days.  Therefore this is not a bug.

**Comment by Linicks** [comment]
@mmahut Thank you for adding to this discussion; your comments are much appreciated and insightful.  I know that the previous peers are part of a breaking change.  The real issue here, as I see it, is that the "test" network is about creating a commonplace to test this software so that it can eventually make it's way to release.  Running a private blockchain is literally missing the point of the entire project and is a futile exercise.  When posting this, I had two objectives.  The first is to add context and documentation to a project that is about as bad as it gets when providing useable documentation. I know that people like me will be trying to migrate their test nodes and will be looking for this information on Google.  Secondly, I'm illustrating the point that these haphazard insider releases are only good for those that are the ones directly touching the code, or close enough to have the information to actually make sense of what's going on.  Yes, "move fast and break things", but this is a project that's aiming to effect allot of people.  Therefore, there should be some consideration to those that want to be part of this community and contribute that aren't insiders.  I will say,  that it's pretty pathetic that both comments here are more about deflecting and pointing fingers than actually providing assistance.    A beneficial comment would have consisted of a solution to the initial report that included a description of how to correct the issue.  "Bugs" aren't always in the actual software, but can also be part of the process.  I know that I've been working hard to jump into the Cardano project because of its ideology and promise.  This software is barely alpha and with the upcoming incentivized testing that's been hyped over and over, I hope that "this" software and community gets it's act together.  IOHK is developing and pushing this, so yes, they should have had docs and a network setup first.  There are professional standards in this industry, open-source or not, that aren't being met.  I hope that adding an outsider's perspective will help you and others that might stumble upon this too reflect on the disconnection of what this project is selling versus what's being delivered.  Cheers!

**Comment by jbax** [comment]
I'm with @Linicks here. Jesus Christ guys please at least add a note saying "don't bother downloading the new RC4 until we have a RC4 network available". And pretty please post the new recommended config with correct peers once that is ready.

**Comment by mmahut** [comment]
I see and understand your point, but I disagree. This is a release candidate (yes, not even a release!) of a blockchain software. I'm using it to run a private network with tests. 

Given this is not a real bug affecting the source code hosted here, Github of the project is not the correct place to inform about a specific testnet setup run by the IOHK. Follow their website, the telegram group or reach their Technical Support.

It's like opening a bug against Apache Foundation that a website on the internet is running an old version of apache httpd. :)

**Comment by mmahut** [comment]
> (...) at least add a note saying "don't bother downloading the new RC4 until we have a RC4 network available".

I think you are missing the point of a decentralized blockchain software. Download it, run it, test it and report any issues you see. More setup scenario's we try, the more stable it will be.

**Comment by jbax** [comment]
@mmahut the documentation that is advertised everywhere tells us to download the latest release, use a config file template provided by IOHK, and start the node. If that is the workflow primarily suggested to the community, then IOHK should warn the community when that won't work anymore.

Is it that hard to simply let everyone know that there is no RC4 network up and running yet on the release notes and prevent people who are not interested in running their own network from wasting time?

**Comment by mmahut** [comment]
> Is it that hard to simply let everyone know that there is no RC4 network up and running yet on the release notes and prevent people who are not interested in running their own network from wasting time?

I think you are missing my point - jormungandr and IOHK's network are two separate things. One is a blockchain, the second is a network ran using that software. Information about the later does not belong to the release notes of the software itself.

There is already a number of RC4 networks running by a number of people, ask around in the telegram channel.

To rephrase: information about someone's network setup (in this case IOHK) does not belong to a release notes of the software itself.

**Comment by jbax** [comment]
Well if you guys want to keep this confusing, then feel free to do so. Don't complain when everyone and their dog open irrelevant github issues that could be prevented simply by communicating instructions **properly**.

**Comment by RonnyTome** [comment]
we are in automatic as soon as they release a new version, we have already installed a new bug search. so it can run with pure accuracy, the next time they release another version include how it should be configured to use the tests. still can't guess

**Comment by vincenthz** [comment]
@mmahut is right here. This is just some software releases which are implied in the name (release candidate) to not be particularly stable/final, which doesn't have to be particularly linked to a network / chain / peers. we put it there as we test it internally, but also have lots of extremely valuable contributions from many external people that made their own test network and lots of other things.

@Linicks @jbax While we don't want to prevent anyone from testing beta releases, it's actually expected to sit-this-one-out until we make proper communications that we'ld like everyone to upgrade, which is not likely to happens before a final 0.7.0. The fact that you didn't find a network / peers / config for rc4 just reinforce that IOHK will not put a stable / published network out until we start the 0.7 "stable" branch properly.

I believe the state of affair of -rc'ness has been communicated recently (i.e. you can help but also should sit this one out if you expect a stable thing), but I'll make sure the comm team is aware of this misunderstanding and clarify this with alacritty.

**Comment by jbax** [comment]
Thanks @vincenthz , I understand what you guys are saying. I just want to make it clear that  people are opening issues and asking around because you guys **are failing to communicate clearly** what everyone is to expect and do on every release (rc or not). Remember the community was invited to join the testnet as if it were "plug and play" so everyone expects it to be like that unless you clearly state that is not the case. 

Don't assume "rc-ness" by itself is clear enough. It. Is. Not.
